x-pack/plugins/infra/public/utils/url_state.tsx

```diff
@@ -155,16 +155,15 @@ export const replaceStateKeyInQueryString = <UrlState extends any>(
   stateKey: string,
   urlState: UrlState | undefined
 ) => (queryString: string) => {
-  const previousQueryValues = parse(queryString, { sort: false });
+  const previousQueryValues = parse(queryString);
   const encodedUrlState =
     typeof urlState !== 'undefined' ? encodeRisonUrlState(urlState) : undefined;
 
   return stringify(
     url.encodeQuery({
       ...previousQueryValues,
       [stateKey]: encodedUrlState,
-    }),
-    { sort: false, encode: false }
```

Comment:
This changes the behaviour of these links (`url.encodeQuery` removing over-zealous encoding, and `stringify` re-adding it).